packages/eas-cli/src/commands/update/list.ts

```diff
@@ -0,0 +1,102 @@
+import { getConfig } from '@expo/config';
+import { Command, flags } from '@oclif/command';
+import Table from 'cli-table3';
+
+import { getUpdates } from '../../../src/update/updateUtils';
+import log from '../../log';
+import { ensureProjectExistsAsync } from '../../project/ensureProjectExists';
+import { findProjectRootAsync, getProjectAccountNameAsync } from '../../project/projectUtils';
+import { promptAsync } from '../../prompts';
+
+export default class UpdateList extends Command {
+  static hidden = true;
+  static description = 'View a list of updates.';
+
+  static args = [
+    {
+      name: 'releaseName',
+      required: false,
+      description: 'Name of the release to view',
+    },
+  ];
+
+  static flags = {
+    platform: flags.string({
+      description: 'Platform-specifc updates to return: ios, android, web.',
+    }),
+    all: flags.boolean({
+      description: 'Return a list of all updates individually.',
+      default: false,
+    }),
+    json: flags.boolean({
+      description: 'Return list of updates as JSON.',
+      default: false,
+    }),
+  };
+
+  async run() {
+    let {
+      args: { releaseName },
+      flags: { json: jsonFlag, platform: platformFlag, all: allFlag },
+    } = this.parse(UpdateList);
+
+    const projectDir = await findProjectRootAsync(process.cwd());
+    if (!projectDir) {
+      throw new Error('Please run this command inside a project directory.');
+    }
+    const accountName = await getProjectAccountNameAsync(projectDir);
+    const {
+      exp: { slug },
+    } = getConfig(projectDir, { skipSDKVersionRequirement: true });
+    const projectId = await ensureProjectExistsAsync({
+      accountName,
+      projectName: slug,
+    });
+
+    if (!releaseName) {
+      const validationMessage = 'Release name may not be empty.';
+      if (jsonFlag) {
+        throw new Error(validationMessage);
+      }
+      ({ releaseName } = await promptAsync({
+        type: 'text',
+        name: 'releaseName',
+        message: 'Please enter the name of the release to view:',
+        validate: value => (value ? true : validationMessage),
+      }));
+    }
+
+    const updates = await getUpdates({
+      projectId,
+      releaseName,
+      platformFlag,
+      allFlag,
+    });
+
+    if (jsonFlag) {
+      log(updates);
```

Comment:
We have this in a few spots around the CLI (to print JSON). I noticed that this can produce a `... ## more` message, like this:

<img width="1086" alt="Screen Shot 2020-12-21 at 12 10 42 PM" src="https://user-images.githubusercontent.com/6455018/102803037-9c7bc300-4385-11eb-868e-99a149ae37ca.png">

Would it be better to use `JSON.stringify(...)`?

Reply:
From what i understand, `--json` is primarily for piping commands so we should be sure that anything we do doesn't affect that. For example, I assume that if we piped this into a file that it would provide the whole JSON?